packages/types/src/CryptoCallbacks.ts

```diff
@@ -6,107 +6,154 @@
  */
 
 import type { SignerPayloadJSON } from '@polkadot/types/types'
+import {
+  DidResourceUri,
+  DidUri,
+  DidVerificationKey,
+  VerificationKeyRelationship,
+} from './DidDocument.js'
 
-export type SigningAlgorithms = 'ed25519' | 'sr25519' | 'ecdsa-secp256k1'
+const signingAlgorithmsC = ['ed25519', 'sr25519', 'ecdsa-secp256k1'] as const
+export const signingAlgorithms = signingAlgorithmsC as unknown as string[]
+export type SigningAlgorithms = typeof signingAlgorithmsC[number]
 
 export type EncryptionAlgorithms = 'x25519-xsalsa20-poly1305'
 
 /**
- * Base interface for all {en/de}cryption & signing requests.
+ * Base interface for all signing requests.
  */
-export interface RequestData<
-  A extends SigningAlgorithms | EncryptionAlgorithms
-> {
+export interface SignRequestData {
   /**
-   * Identifier for the encryption/signing algorithm to use.
+   * Data to be signed.
    */
-  alg: A
+  data: Uint8Array
   /**
-   * Public key as u8a identifying the keypair to use (in combination with the alg).
+   * Info for extensions to display to user.
+   * Available when signing extrinsics.
    */
-  publicKey: Uint8Array
+  meta?: Partial<SignerPayloadJSON>
```

Comment:
what do you think, should we rename this parameter? I think it's called meta on the polkadot-js signer interface but our callback doesn't really have anything to do with it anyway, I suppose. So a more descriptive name probably won't hurt

Reply:
ah ok. I only know it as meta. any suggestions?

Reply:
we were asking ourselves, if we need it even. Will have a look.